controllers/ingredients.js

```diff
@@ -2,9 +2,9 @@ import Ingredients from "../models/ingredients.js";
 
 export async function getAllIngredients(req, res, next) {
   try {
-    const { page = 1, limit = 10 } = req.query;
-    const findPromise = Ingredients.find({}, {}, { skip: limit * (page - 1), limit });
-    const countPromise = Ingredients.countDocuments();
+    const { page = 1, limit = 10, q = "" } = req.query;
+    const findPromise = Ingredients.find({ name: { $regex: q, $options: "i" } }, {}, { skip: limit * (page - 1), limit });
+    const countPromise = Ingredients.countDocuments({ name: { $regex: q, $options: "i" } });

```

Comment:
Make one common variable for filter object.

Reply:
> Make one common variable for filter object.

done.